File: backend/app/schemas/employee.py
from datetime import date, datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr

from app.models.employee import EmploymentType


class EmployeeCreate(BaseModel):
    full_name: str
    job_title: str
    country: str
    salary: Decimal
    department: str
    email: EmailStr
    hire_date: date
    employment_type: EmploymentType = EmploymentType.FULL_TIME


class EmployeeUpdate(BaseModel):
    full_name: Optional[str] = None
    job_title: Optional[str] = None
    country: Optional[str] = None
    salary: Optional[Decimal] = None
    department: Optional[str] = None
    email: Optional[EmailStr] = None
    hire_date: Optional[date] = None
    employment_type: Optional[EmploymentType] = None


class EmployeeResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    full_name: str
    job_title: str
    country: str
    salary: Decimal
    department: str
    email: str
    hire_date: date
    employment_type: str
    is_active: bool
    created_at: datetime
    updated_at: datetime


class EmployeeListResponse(BaseModel):
    items: list[EmployeeResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/employee.ts
export type EmploymentType = 'full_time' | 'part_time' | 'contract';

export interface Employee {
  id: string;
  full_name: string;
  job_title: string;
  country: string;
  salary: number;
  department: string;
  email: string;
  hire_date: string;
  employment_type: EmploymentType;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface EmployeeCreate {
  full_name: string;
  job_title: string;
  country: string;
  salary: number;
  department: string;
  email: string;
  hire_date: string;
  employment_type: EmploymentType;
}

export type EmployeeUpdate = Partial<EmployeeCreate>;

export interface EmployeeListResponse {
  items: Employee[];
  total: number;
  page: number;
  page_size: number;
}

export interface EmployeeListParams {
  page?: number;
  page_size?: number;
  country?: string;
  job_title?: string;
  department?: string;
  search?: string;
  sort_by?: string;
  sort_order?: 'asc' | 'desc';
}
